test/e2e/flowaggregator_test.go

```diff
@@ -0,0 +1,335 @@
+						t.Fatalf("Records does not have correct destinationServicePortName")
+					}
+				}
+			} else {
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "destinationPodName", "perftest-c")) {
+					t.Fatalf("Records does not have correct destinationPodName")
+				}
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "destinationNodeName", workerNodeName(1))) {
+					t.Fatalf("Records does not have correct destinationNodeName")
+				}
+				if checkService {
+					if !strings.Contains(record, fmt.Sprintf("%s: %s", "destinationServicePortName", "antrea-test/perftest-c")) {
+						t.Fatalf("Records does not have correct destinationServicePortName")
+					}
+				}
+			}
+			if checkNetworkPolicy {
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "ingressNetworkPolicyName", "test-networkpolicy-ingress")) {
+					t.Fatalf("Records does not have correct ingressNetworkPolicyName")
+				}
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "ingressNetworkPolicyNamespace", testNamespace)) {
+					t.Fatalf("Records does not have correct ingressNetworkPolicyNamespace")
+				}
+			}
+		}
+	}
+	t.Logf("Receive template record number: %d, data record number: %d", templateRecords, dataRecordsCount)
+	assert.Equal(t, templateRecords, 1, "Flow aggregator should send out 1 template record")
+	// Single iperf resulting in two connections with separate ports. Suspecting second flow to be control flow to exchange
+	// stats info. As 5s is export interval and iperf traffic runs for 10s, we expect about 4 records exporting to the flow aggregator.
+	// Since flow aggregator will aggregate records based on 5-tuple connection key, so we expect 2 records.
+	assert.GreaterOrEqual(t, dataRecordsCount, 2, "Iperf flow should have expected number of flow records")
+}
+
+func getRecordsFromOutput(output string) []string {
+	re := regexp.MustCompile("(?m)^.*" + "#" + ".*$[\r\n]+")
+	output = re.ReplaceAllString(output, "")
+	output = strings.TrimSpace(output)
+	recordSlices := strings.Split(output, "IPFIX-HDR:")
+	// Delete the first element from recordSlices
+	recordSlices[0] = recordSlices[len(recordSlices)-1]
+	recordSlices[len(recordSlices)-1] = ""
+	recordSlices = recordSlices[:len(recordSlices)-1]
+	return recordSlices
+}
+
+func deployNetworkPolicy(t *testing.T, data *TestData) (np1 *networkingv1.NetworkPolicy, np2 *networkingv1.NetworkPolicy) {
+	// Add NetworkPolicy between two iperf Pods.
+	var err error
+	np1, err = data.createNetworkPolicy("test-networkpolicy-ingress", &networkingv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{},
+		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		Ingress: []networkingv1.NetworkPolicyIngressRule{{
+			From: []networkingv1.NetworkPolicyPeer{{
+				PodSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"antrea-e2e": "perftest-a",
+					},
+				}},
+			},
+		}},
+	})
+	if err != nil {
+		t.Fatalf("Error when creating Network Policy: %v", err)
```

Comment:
Please change fatal message to error as master branch to make sure the test deployments are teared down before exiting. Same for most of the error logs in this file.

Reply:
Thanks, addressed.